actionview/test/template/dependency_tracker_test.rb

```diff
@@ -48,69 +55,72 @@ def test_returns_empty_array_if_no_tracker_is_found
   end
 end
 
-class ERBTrackerTest < Minitest::Test
+class RipperTrackerTest < Minitest::Test
   def make_tracker(name, template)
-    ActionView::DependencyTracker::ERBTracker.new(name, template)
+    ActionView::DependencyTracker::RipperTracker.new(name, template)
   end
 
   def test_dependency_of_erb_template_with_number_in_filename
-    template = FakeTemplate.new("<%# render 'messages/message123' %>", :erb)
+    template = FakeTemplate.new("<%= render 'messages/message123' %>", :erb)
     tracker = make_tracker("messages/_message123", template)
 
     assert_equal ["messages/message123"], tracker.dependencies
   end
 
   def test_dependency_of_template_partial_with_layout
-    template = FakeTemplate.new("<%# render partial: 'messages/show', layout: 'messages/layout' %>", :erb)
+    template = FakeTemplate.new("<%= render partial: 'messages/show', layout: 'messages/layout' %>", :erb)
     tracker = make_tracker("multiple/_dependencies", template)
 
-    assert_equal ["messages/layout", "messages/show"], tracker.dependencies
+    assert_equal ["messages/show", "messages/layout"], tracker.dependencies
```

Comment:
Why was the order swapped?

Reply:
Yeah, ripper can tell when code is a comment and skips renders that are commented out. I added a new test for this `test_dependencies_skip_commented_out_renders`
https://github.com/rails/rails/pull/42458/files#diff-30b8d80d32e6ac1253f719c5204ba33a22e41cef79edfce28a7637df65606ac1R216-R223